Add Google Cloud Platform Search extension

klaa97 opened the pull request:
## Description

This PR adds an extension to search projects and products in Google Cloud Platform, and open them in the browser (either in the documentation, or in the Google Cloud Console).

To search and update projects, the Resource Manager API is used; this requires the `gcloud cli` to be installed, as written in the README. Although this makes it necessary to have an external binary:
- It's quite likely that people interested in GCP shortcuts to access projects have the SDK installed
- It avoid the needs of specific secret keys to use, reducing the trust needed to save keys in preferences - instead, it just uses the official library which is authenticated through the CLI

I added in the README more information about of all this; an alternative would be to ask a specific key and save in the preferences, but i'd avoid if possible at least initially. Let me know what you think!

(For reference, this takes a bit of inspiration from the great plugin for Alfed: https://github.com/jarlefosen/alfred-gcloud-shortcuts and enhance it by adding more products, and the documentation command).

Possible improvements would be to add product-specific functions (such as check number of machines, buckets, and so on), or use a frecency algorithm to list the items; as a start though, it would improve a lot my workflow and make me go full on Raycast personally 😄  

## Type of change

- New extension

## Screencast

<!-- If you add a new extension or command, include a screencast (or screenshot for a small change). Good screencast will make the review much faster, especially your extension requires registration in other services.  -->


https://user-images.githubusercontent.com/39653693/151674027-a7ca1885-fcc9-480f-831c-d30f41814c33.mp4



## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)

pernielsentikaer reviewed (changes requested):
Hi @klaa97 👋 

Thanks for your awesome contribution 💪 I don't use it myself, so it's quite hard to test out.

Some feedback for you: 

- Instead of using node-localstorage you could use the one that's [built-in](https://developers.raycast.com/api-reference/storage#localstoragevalues) within Raycast if I'm not mistaken

- Since the README button only is visible if you have a required setting, it's quite hard to find out how to use it. An idea could be to use something like a `<Detail markdown="">` and “build” your own with header, image, and some explanation.

Does the above make sense?

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

klaa97 commented:
Thank you for your review @pernielsentikaer !

- I refactored a bit the code to use the Raycast provided APIs, and removing some dependencies that were not useful (such as SWR)
- I added a detailed explanation of the command in a dedicated message which is shown when the command fails to run and there is no cache: this allows the user to check how to configure it easily

Let me know what you think, thank you again 💪

pernielsentikaer reviewed (approved):
Hi @klaa97 👋 

Well done, it looks very good and good job with the new Detail views! 🔥 

I don't have it myself, so I can't test it as much as I wanted to, but I have checked all the code, and it looks good 😄 

I'm going to accept it and then we await final review before merging 🚢
